src/middleware/oauthServer.ts

```diff
@@ -0,0 +1,115 @@
+import { Request, Response } from 'express';
+import { Request as OAuth2Request, Response as OAuth2Response } from '@node-oauth/oauth2-server';
+import OAuth2Server from '@node-oauth/oauth2-server';
+import { OAuthClient } from '../models/OAuthClient';
+import { OAuthCode } from '../models/OAuthCode';
+import { OAuthRefreshToken } from '../models/OAuthRefreshToken';
+import { OAuthToken } from '../models/OAuthToken';
+import { User } from '../models/User';
+import { AuthorizationCode, Client, Token } from 'oauth2-server'
+
+const model = {
+  // Fetch the OAuth2 client from the DB
+  getClient: async function(clientId: string, clientSecret: string) {
+    return await OAuthClient.findOne(clientId, clientSecret);
+  },
+
+  // Fetch the authorization code from the DB
+  getAuthorizationCode: async function(authorizationCode: string) {
+    return await OAuthCode.findOne(authorizationCode);
+  },
+
+  // Save the authorization code in the DB
+  saveAuthorizationCode: async function(code: AuthorizationCode, client: Client, user: User) {
+    try {
+      code.client = client;
+      code.user = user;
+      return await code.save();
+    } catch(err) {
+      console.log(err);
+      return null;
+    }
+  },
+
+  // Set the expiry date on the authorization code to the current time
+  revokeAuthorizationCode: async function(code: AuthorizationCode) {
+    try {
+      return await code.revoke();
+    } catch(err) {
+      console.log(err);
+      return null;
+    }
+  },
+
+  // Verify that the scope requested is allowed by the access token
+  verifyScope: async function(token: Token, scope: string | string[]) {
+    const scopeArray: string[] = Array.isArray(scope) ? scope : [scope];
+    return await scopeArray.every((item) => token.scope.includes(item));
```

Comment:
L47: This looks like a synchronous function. I don't think you need the "`await`"